Build failure: hatch

## Steps To Reproduce

Steps to reproduce the behavior:

1. `nix-shell -p hatch`

## Build log

<details>

<summary>Build Log</summary>

```
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pypa-build-hook
Using pypaBuildPhase
Sourcing python-runtime-deps-check-hook
Using pythonRuntimeDepsCheckHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing pytest-check-hook
Using pytestCheckPhase
Using versionCheckHook
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/s6467b3pw6rjlz9xw22jradshpr17ylv-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/tests/venv/test_utils.py
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/ncc382zvq6qpn0b1gmr02b2k3slik21s-paths.patch
patching file src/hatch/env/virtual.py
patching file src/hatch/venv/core.py
patching file tests/conftest.py
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing pypaBuildPhase
Setting SETUPTOOLS_SCM_PRETEND_VERSION to 1.13.0
Including all tracked files automatically
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
[1m* Getting build dependencies for wheel...[0m
[1m* Building wheel...[0m
[1m[92mSuccessfully built [4mhatch-1.13.0-py3-none-any.whl[0m[1m[92m[0m
Finished creating a wheel...
/private/tmp/nix-build-hatch-1.13.0.drv-0/source/dist /private/tmp/nix-build-hatch-1.13.0.drv-0/source
Unpacking to: unpacked/hatch-1.13.0...OK
Repacking wheel as ./hatch-1.13.0-py3-none-any.whl...OK
/private/tmp/nix-build-hatch-1.13.0.drv-0/source
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
@nix { "action": "setPhase", "phase": "pythonRuntimeDepsCheckHook" }
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for hatch-1.13.0-py3-none-any.whl
Finished executing pythonRuntimeDepsCheck
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Executing pypaInstallPhase
Successfully installed hatch-1.13.0-py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
@nix { "action": "setPhase", "phase": "pythonOutputDistPhase" }
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
checking for references to /private/tmp/nix-build-hatch-1.13.0.drv-0/ in /nix/store/k4gwd3yna8d297jsrip6wrvy41r0nq97-hatch-1.13.0...
patching script interpreter paths in /nix/store/k4gwd3yna8d297jsrip6wrvy41r0nq97-hatch-1.13.0
stripping (with command strip and flags -S) in  /nix/store/k4gwd3yna8d297jsrip6wrvy41r0nq97-hatch-1.13.0/lib /nix/store/k4gwd3yna8d297jsrip6wrvy41r0nq97-hatch-1.13.0/bin
checking for references to /private/tmp/nix-build-hatch-1.13.0.drv-0/ in /nix/store/xcv4sv7vwxxvk2nmlipiv9ml4ynzy6sm-hatch-1.13.0-dist...
patching script interpreter paths in /nix/store/xcv4sv7vwxxvk2nmlipiv9ml4ynzy6sm-hatch-1.13.0-dist
Rewriting #!/nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12 to #!/nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7
wrapping `/nix/store/k4gwd3yna8d297jsrip6wrvy41r0nq97-hatch-1.13.0/bin/hatch'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
Executing versionCheckPhase
Successfully managed to find version 1.13.0 in the output of the command /nix/store/k4gwd3yna8d297jsrip6wrvy41r0nq97-hatch-1.13.0/bin/hatch --version
Hatch, version 1.13.0
Finished versionCheckPhase
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
@nix { "action": "setPhase", "phase": "pythonCatchConflictsPhase" }
Running phase: pythonRemoveBinBytecodePhase
@nix { "action": "setPhase", "phase": "pythonRemoveBinBytecodePhase" }
Running phase: pythonImportsCheckPhase
@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
Executing pythonImportsCheckPhase
Running phase: pytestXdistHook
@nix { "action": "setPhase", "phase": "pytestXdistHook" }
Running phase: pytestCheckPhase
@nix { "action": "setPhase", "phase": "pytestCheckPhase" }
Executing pytestCheckPhase
[1m============================= test session starts ==============================[0m
platform darwin -- Python 3.12.7, pytest-8.3.3, pluggy-1.5.0
rootdir: /private/tmp/nix-build-hatch-1.13.0.drv-0/source
configfile: pyproject.toml
plugins: mock-3.14.0, xdist-3.6.1, anyio-4.6.2
[1m
created: 14/14 workers[0m[1m
initialized: 1/14 workers[0m[1m
initialized: 2/14 workers[0m[1m
initialized: 3/14 workers[0m[1m
initialized: 4/14 workers[0m[1m
initialized: 5/14 workers[0m[1m
initialized: 6/14 workers[0m[1m
initialized: 7/14 workers[0m[1m
initialized: 8/14 workers[0m[1m
initialized: 9/14 workers[0m[1m
initialized: 10/14 workers[0m[1m
initialized: 11/14 workers[0m[1m
initialized: 12/14 workers[0m[1m
initialized: 13/14 workers[0m[1m
initialized: 14/14 workers[0m[1m
ready: 1/14 workers       [0m[1m
ready: 2/14 workers[0m[1m
ready: 3/14 workers[0m[1m
ready: 4/14 workers[0m[1m
ready: 5/14 workers[0m[1m
ready: 6/14 workers[0m[1m
ready: 7/14 workers[0m[1m
ready: 8/14 workers[0m[1m
ready: 9/14 workers[0m[1m
ready: 10/14 workers[0m[1m
ready: 11/14 workers[0m[1m
ready: 12/14 workers[0m[1m
ready: 13/14 workers[0m[1m
ready: 14/14 workers[0m[1m
collecting: 1/14 workers[0m[1m
collecting: 2/14 workers[0m[1m
collecting: 3/14 workers[0m[1m
collecting: 4/14 workers[0m[1m
collecting: 5/14 workers[0m[1m
collecting: 6/14 workers[0m[1m
collecting: 7/14 workers[0m[1m
collecting: 8/14 workers[0m[1m
collecting: 9/14 workers[0m[1m
collecting: 10/14 workers[0m[1m
collecting: 11/14 workers[0m[1m
collecting: 12/14 workers[0m[1m
collecting: 13/14 workers[0m[1m
14 workers [1553 items]  [0m
[32m.[0m<snip>                            [100%][0m
=================================== FAILURES ===================================
[31m[1m______ TestBuildDependenciesMissing.test_no_compatibility_check_if_exists ______[0m
[gw8] darwin -- Python 3.12.7 /nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12

self = <tests.cli.project.test_metadata.TestBuildDependenciesMissing object at 0x11c0d56a0>
hatch = <tests.conftest.CliRunner object at 0x11cfdb170>
temp_dir = Path('/private/tmp/nix-build-hatch-1.13.0.drv-0/tmpihtfet4v')
helpers = <module 'tests.helpers.helpers' from '/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/helpers/helpers.py'>
mocker = <pytest_mock.plugin.MockerFixture object at 0x11cfd6870>

    [0m[37m@pytest[39;49;00m.mark.allow_backend_process[90m[39;49;00m
    [37m@pytest[39;49;00m.mark.requires_internet[90m[39;49;00m
    [94mdef[39;49;00m [92mtest_no_compatibility_check_if_exists[39;49;00m([96mself[39;49;00m, hatch, temp_dir, helpers, mocker):[90m[39;49;00m
        project_name = [33m'[39;49;00m[33mMy.App[39;49;00m[33m'[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m temp_dir.as_cwd():[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33mnew[39;49;00m[33m'[39;49;00m, project_name)[90m[39;49;00m
            [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
    [90m[39;49;00m
        project_path = temp_dir / [33m'[39;49;00m[33mmy-app[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path = temp_dir / [33m'[39;49;00m[33mdata[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path.mkdir()[90m[39;49;00m
    [90m[39;49;00m
        project = Project(project_path)[90m[39;49;00m
        config = [96mdict[39;49;00m(project.raw_config)[90m[39;49;00m
        config[[33m'[39;49;00m[33mbuild-system[39;49;00m[33m'[39;49;00m][[33m'[39;49;00m[33mrequires[39;49;00m[33m'[39;49;00m].append([33m'[39;49;00m[33mbinary[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
        project.save_config(config)[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m project_path.as_cwd(env_vars={ConfigEnvVars.DATA: [96mstr[39;49;00m(data_path)}):[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33mproject[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mmetadata[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mlicense[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
>       [94massert[39;49;00m result.output == helpers.dedent([90m[39;49;00m
    [90m        [39;49;00m[33m"""[39;49;00m
    [33m        Setting up build environment for missing dependencies[39;49;00m
    [33m        MIT[39;49;00m
    [33m        """[39;49;00m[90m[39;49;00m
        )[90m[39;49;00m
[1m[31mE       AssertionError: assert 'MIT\n' == 'Setting up b...encies\nMIT\n'[0m
[1m[31mE         [0m
[1m[31mE         - Setting up build environment for missing dependencies[0m
[1m[31mE           MIT[0m

[1m[31m/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/cli/project/test_metadata.py[0m:406: AssertionError
[31m[1m____________________________ test_sync_dependencies ____________________________[0m
[gw8] darwin -- Python 3.12.7 /nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12

hatch = <tests.conftest.CliRunner object at 0x11cfdb170>
helpers = <module 'tests.helpers.helpers' from '/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/helpers/helpers.py'>
temp_dir = Path('/private/tmp/nix-build-hatch-1.13.0.drv-0/tmplta3p_r5')
config_file = <hatch.config.user.ConfigFile object at 0x11cc5d6d0>

    [0m[37m@pytest[39;49;00m.mark.requires_internet[90m[39;49;00m
    [94mdef[39;49;00m [92mtest_sync_dependencies[39;49;00m(hatch, helpers, temp_dir, config_file):[90m[39;49;00m
        config_file.model.template.plugins[[33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m][[33m'[39;49;00m[33mtests[39;49;00m[33m'[39;49;00m] = [94mFalse[39;49;00m[90m[39;49;00m
        config_file.save()[90m[39;49;00m
    [90m[39;49;00m
        project_name = [33m'[39;49;00m[33mMy.App[39;49;00m[33m'[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m temp_dir.as_cwd():[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33mnew[39;49;00m[33m'[39;49;00m, project_name)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
    [90m[39;49;00m
        project_path = temp_dir / [33m'[39;49;00m[33mmy-app[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path = temp_dir / [33m'[39;49;00m[33mdata[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path.mkdir()[90m[39;49;00m
    [90m[39;49;00m
        project = Project(project_path)[90m[39;49;00m
        helpers.update_project_environment(project, [33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m, {[33m'[39;49;00m[33mskip-install[39;49;00m[33m'[39;49;00m: [94mTrue[39;49;00m, **project.config.envs[[33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m]})[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m project_path.as_cwd(env_vars={ConfigEnvVars.DATA: [96mstr[39;49;00m(data_path)}):[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33menv[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mcreate[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
        [94massert[39;49;00m result.output == helpers.dedent([90m[39;49;00m
    [90m        [39;49;00m[33m"""[39;49;00m
    [33m        Creating environment: default[39;49;00m
    [33m        Checking dependencies[39;49;00m
    [33m        """[39;49;00m[90m[39;49;00m
        )[90m[39;49;00m
    [90m[39;49;00m
        env_data_path = data_path / [33m'[39;49;00m[33menv[39;49;00m[33m'[39;49;00m / [33m'[39;49;00m[33mvirtual[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        [94massert[39;49;00m env_data_path.is_dir()[90m[39;49;00m
    [90m[39;49;00m
        project_data_path = env_data_path / project_path.name[90m[39;49;00m
        [94massert[39;49;00m project_data_path.is_dir()[90m[39;49;00m
    [90m[39;49;00m
        storage_dirs = [96mlist[39;49;00m(project_data_path.iterdir())[90m[39;49;00m
        [94massert[39;49;00m [96mlen[39;49;00m(storage_dirs) == [94m1[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        storage_path = storage_dirs[[94m0[39;49;00m][90m[39;49;00m
        [94massert[39;49;00m [96mlen[39;49;00m(storage_path.name) == [94m8[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        env_dirs = [96mlist[39;49;00m(storage_path.iterdir())[90m[39;49;00m
        [94massert[39;49;00m [96mlen[39;49;00m(env_dirs) == [94m1[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        env_path = env_dirs[[94m0[39;49;00m][90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m env_path.name == project_path.name[90m[39;49;00m
    [90m[39;49;00m
        project = Project(project_path)[90m[39;49;00m
        helpers.update_project_environment([90m[39;49;00m
            project, [33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m, {[33m'[39;49;00m[33mdependencies[39;49;00m[33m'[39;49;00m: [[33m'[39;49;00m[33mbinary[39;49;00m[33m'[39;49;00m], **project.config.envs[[33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m]}[90m[39;49;00m
        )[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m project_path.as_cwd(env_vars={ConfigEnvVars.DATA: [96mstr[39;49;00m(data_path)}):[90m[39;49;00m
            result = hatch([90m[39;49;00m
                [33m'[39;49;00m[33mrun[39;49;00m[33m'[39;49;00m,[90m[39;49;00m
                [33m'[39;49;00m[33mpython[39;49;00m[33m'[39;49;00m,[90m[39;49;00m
                [33m'[39;49;00m[33m-c[39;49;00m[33m'[39;49;00m,[90m[39;49;00m
                [33m"[39;49;00m[33mimport binary,pathlib,sys;pathlib.Path([39;49;00m[33m'[39;49;00m[33mtest.txt[39;49;00m[33m'[39;49;00m[33m).write_text(str(binary.convert_units(1024)))[39;49;00m[33m"[39;49;00m,[90m[39;49;00m
            )[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
>       [94massert[39;49;00m result.output == helpers.dedent([90m[39;49;00m
    [90m        [39;49;00m[33m"""[39;49;00m
    [33m        Checking dependencies[39;49;00m
    [33m        Syncing dependencies[39;49;00m
    [33m        """[39;49;00m[90m[39;49;00m
        )[90m[39;49;00m
[1m[31mE       AssertionError: assert 'Checking dependencies\n' == 'Checking dep...ependencies\n'[0m
[1m[31mE         [0m
[1m[31mE           Checking dependencies[0m
[1m[31mE         - Syncing dependencies[0m

[1m[31m/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/cli/run/test_run.py[0m:273: AssertionError
[31m[1m____________________ test_no_compatibility_check_if_exists _____________________[0m
[gw4] darwin -- Python 3.12.7 /nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12

hatch = <tests.conftest.CliRunner object at 0x113f6d160>
helpers = <module 'tests.helpers.helpers' from '/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/helpers/helpers.py'>
temp_dir = Path('/private/tmp/nix-build-hatch-1.13.0.drv-0/tmpu2xjhag0')
mocker = <pytest_mock.plugin.MockerFixture object at 0x115f35a30>

    [0m[37m@pytest[39;49;00m.mark.requires_internet[90m[39;49;00m
    [94mdef[39;49;00m [92mtest_no_compatibility_check_if_exists[39;49;00m(hatch, helpers, temp_dir, mocker):[90m[39;49;00m
        project_name = [33m'[39;49;00m[33mMy.App[39;49;00m[33m'[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m temp_dir.as_cwd():[90m[39;49;00m
            hatch([33m'[39;49;00m[33mnew[39;49;00m[33m'[39;49;00m, project_name)[90m[39;49;00m
    [90m[39;49;00m
        project_path = temp_dir / [33m'[39;49;00m[33mmy-app[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path = temp_dir / [33m'[39;49;00m[33mdata[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path.mkdir()[90m[39;49;00m
    [90m[39;49;00m
        project = Project(project_path)[90m[39;49;00m
        config = [96mdict[39;49;00m(project.raw_config)[90m[39;49;00m
        config[[33m'[39;49;00m[33mbuild-system[39;49;00m[33m'[39;49;00m][[33m'[39;49;00m[33mrequires[39;49;00m[33m'[39;49;00m].append([33m'[39;49;00m[33mbinary[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
        project.save_config(config)[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m project_path.as_cwd(env_vars={ConfigEnvVars.DATA: [96mstr[39;49;00m(data_path)}):[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33mversion[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
>       [94massert[39;49;00m result.output == helpers.dedent([90m[39;49;00m
    [90m        [39;49;00m[33m"""[39;49;00m
    [33m        Setting up build environment for missing dependencies[39;49;00m
    [33m        """[39;49;00m[90m[39;49;00m
        )[90m[39;49;00m
[1m[31mE       AssertionError: assert '0.0.1\n' == 'Setting up b...ependencies\n'[0m
[1m[31mE         [0m
[1m[31mE         - Setting up build environment for missing dependencies[0m
[1m[31mE         + 0.0.1[0m

[1m[31m/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/cli/version/test_version.py[0m:181: AssertionError
[31m[1m________________________ test_sync_project_dependencies ________________________[0m
[gw8] darwin -- Python 3.12.7 /nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12

hatch = <tests.conftest.CliRunner object at 0x11cfdb170>
helpers = <module 'tests.helpers.helpers' from '/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/helpers/helpers.py'>
temp_dir = Path('/private/tmp/nix-build-hatch-1.13.0.drv-0/tmps4o3kvku')
config_file = <hatch.config.user.ConfigFile object at 0x11bb97050>

    [0m[37m@pytest[39;49;00m.mark.requires_internet[90m[39;49;00m
    [94mdef[39;49;00m [92mtest_sync_project_dependencies[39;49;00m(hatch, helpers, temp_dir, config_file):[90m[39;49;00m
        config_file.model.template.plugins[[33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m][[33m'[39;49;00m[33mtests[39;49;00m[33m'[39;49;00m] = [94mFalse[39;49;00m[90m[39;49;00m
        config_file.save()[90m[39;49;00m
    [90m[39;49;00m
        project_name = [33m'[39;49;00m[33mMy.App[39;49;00m[33m'[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m temp_dir.as_cwd():[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33mnew[39;49;00m[33m'[39;49;00m, project_name)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
    [90m[39;49;00m
        project_path = temp_dir / [33m'[39;49;00m[33mmy-app[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path = temp_dir / [33m'[39;49;00m[33mdata[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path.mkdir()[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m project_path.as_cwd(env_vars={ConfigEnvVars.DATA: [96mstr[39;49;00m(data_path)}):[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33menv[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mcreate[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
        [94massert[39;49;00m result.output == helpers.dedent([90m[39;49;00m
    [90m        [39;49;00m[33m"""[39;49;00m
    [33m        Creating environment: default[39;49;00m
    [33m        Installing project in development mode[39;49;00m
    [33m        Checking dependencies[39;49;00m
    [33m        """[39;49;00m[90m[39;49;00m
        )[90m[39;49;00m
    [90m[39;49;00m
        env_data_path = data_path / [33m'[39;49;00m[33menv[39;49;00m[33m'[39;49;00m / [33m'[39;49;00m[33mvirtual[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        [94massert[39;49;00m env_data_path.is_dir()[90m[39;49;00m
    [90m[39;49;00m
        project_data_path = env_data_path / project_path.name[90m[39;49;00m
        [94massert[39;49;00m project_data_path.is_dir()[90m[39;49;00m
    [90m[39;49;00m
        storage_dirs = [96mlist[39;49;00m(project_data_path.iterdir())[90m[39;49;00m
        [94massert[39;49;00m [96mlen[39;49;00m(storage_dirs) == [94m1[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        storage_path = storage_dirs[[94m0[39;49;00m][90m[39;49;00m
        [94massert[39;49;00m [96mlen[39;49;00m(storage_path.name) == [94m8[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        env_dirs = [96mlist[39;49;00m(storage_path.iterdir())[90m[39;49;00m
        [94massert[39;49;00m [96mlen[39;49;00m(env_dirs) == [94m1[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        env_path = env_dirs[[94m0[39;49;00m][90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m env_path.name == project_path.name[90m[39;49;00m
    [90m[39;49;00m
        project = Project(project_path)[90m[39;49;00m
        config = [96mdict[39;49;00m(project.raw_config)[90m[39;49;00m
        config[[33m'[39;49;00m[33mproject[39;49;00m[33m'[39;49;00m][[33m'[39;49;00m[33mdependencies[39;49;00m[33m'[39;49;00m] = [[33m'[39;49;00m[33mbinary[39;49;00m[33m'[39;49;00m][90m[39;49;00m
        project.save_config(config)[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m project_path.as_cwd(env_vars={ConfigEnvVars.DATA: [96mstr[39;49;00m(data_path)}):[90m[39;49;00m
            result = hatch([90m[39;49;00m
                [33m'[39;49;00m[33mrun[39;49;00m[33m'[39;49;00m,[90m[39;49;00m
                [33m'[39;49;00m[33mpython[39;49;00m[33m'[39;49;00m,[90m[39;49;00m
                [33m'[39;49;00m[33m-c[39;49;00m[33m'[39;49;00m,[90m[39;49;00m
                [33m"[39;49;00m[33mimport binary,pathlib,sys;pathlib.Path([39;49;00m[33m'[39;49;00m[33mtest.txt[39;49;00m[33m'[39;49;00m[33m).write_text(str(binary.convert_units(1024)))[39;49;00m[33m"[39;49;00m,[90m[39;49;00m
            )[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
>       [94massert[39;49;00m result.output == helpers.dedent([90m[39;49;00m
    [90m        [39;49;00m[33m"""[39;49;00m
    [33m        Checking dependencies[39;49;00m
    [33m        Syncing dependencies[39;49;00m
    [33m        """[39;49;00m[90m[39;49;00m
        )[90m[39;49;00m
[1m[31mE       AssertionError: assert 'Checking dependencies\n' == 'Checking dep...ependencies\n'[0m
[1m[31mE         [0m
[1m[31mE           Checking dependencies[0m
[1m[31mE         - Syncing dependencies[0m

[1m[31m/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/cli/run/test_run.py[0m:346: AssertionError
[31m[1m__________________________ test_sync_project_features __________________________[0m
[gw8] darwin -- Python 3.12.7 /nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12

hatch = <tests.conftest.CliRunner object at 0x11cfdb170>
helpers = <module 'tests.helpers.helpers' from '/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/helpers/helpers.py'>
temp_dir = Path('/private/tmp/nix-build-hatch-1.13.0.drv-0/tmpx1rpqle5')
config_file = <hatch.config.user.ConfigFile object at 0x11cc5f6b0>

    [0m[37m@pytest[39;49;00m.mark.requires_internet[90m[39;49;00m
    [94mdef[39;49;00m [92mtest_sync_project_features[39;49;00m(hatch, helpers, temp_dir, config_file):[90m[39;49;00m
        config_file.model.template.plugins[[33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m][[33m'[39;49;00m[33mtests[39;49;00m[33m'[39;49;00m] = [94mFalse[39;49;00m[90m[39;49;00m
        config_file.save()[90m[39;49;00m
    [90m[39;49;00m
        project_name = [33m'[39;49;00m[33mMy.App[39;49;00m[33m'[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m temp_dir.as_cwd():[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33mnew[39;49;00m[33m'[39;49;00m, project_name)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
    [90m[39;49;00m
        project_path = temp_dir / [33m'[39;49;00m[33mmy-app[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path = temp_dir / [33m'[39;49;00m[33mdata[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path.mkdir()[90m[39;49;00m
    [90m[39;49;00m
        project = Project(project_path)[90m[39;49;00m
        config = [96mdict[39;49;00m(project.raw_config)[90m[39;49;00m
        config[[33m'[39;49;00m[33mproject[39;49;00m[33m'[39;49;00m][[33m'[39;49;00m[33moptional-dependencies[39;49;00m[33m'[39;49;00m] = {[33m'[39;49;00m[33mfoo[39;49;00m[33m'[39;49;00m: []}[90m[39;49;00m
        project.save_config(config)[90m[39;49;00m
        helpers.update_project_environment(project, [33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m, {[33m'[39;49;00m[33mfeatures[39;49;00m[33m'[39;49;00m: [[33m'[39;49;00m[33mfoo[39;49;00m[33m'[39;49;00m], **project.config.envs[[33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m]})[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m project_path.as_cwd(env_vars={ConfigEnvVars.DATA: [96mstr[39;49;00m(data_path)}):[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33menv[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mcreate[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
        [94massert[39;49;00m result.output == helpers.dedent([90m[39;49;00m
    [90m        [39;49;00m[33m"""[39;49;00m
    [33m        Creating environment: default[39;49;00m
    [33m        Installing project in development mode[39;49;00m
    [33m        Checking dependencies[39;49;00m
    [33m        """[39;49;00m[90m[39;49;00m
        )[90m[39;49;00m
    [90m[39;49;00m
        env_data_path = data_path / [33m'[39;49;00m[33menv[39;49;00m[33m'[39;49;00m / [33m'[39;49;00m[33mvirtual[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        [94massert[39;49;00m env_data_path.is_dir()[90m[39;49;00m
    [90m[39;49;00m
        project_data_path = env_data_path / project_path.name[90m[39;49;00m
        [94massert[39;49;00m project_data_path.is_dir()[90m[39;49;00m
    [90m[39;49;00m
        storage_dirs = [96mlist[39;49;00m(project_data_path.iterdir())[90m[39;49;00m
        [94massert[39;49;00m [96mlen[39;49;00m(storage_dirs) == [94m1[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        storage_path = storage_dirs[[94m0[39;49;00m][90m[39;49;00m
        [94massert[39;49;00m [96mlen[39;49;00m(storage_path.name) == [94m8[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        env_dirs = [96mlist[39;49;00m(storage_path.iterdir())[90m[39;49;00m
        [94massert[39;49;00m [96mlen[39;49;00m(env_dirs) == [94m1[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        env_path = env_dirs[[94m0[39;49;00m][90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m env_path.name == project_path.name[90m[39;49;00m
    [90m[39;49;00m
        project = Project(project_path)[90m[39;49;00m
        config = [96mdict[39;49;00m(project.raw_config)[90m[39;49;00m
        config[[33m'[39;49;00m[33mproject[39;49;00m[33m'[39;49;00m][[33m'[39;49;00m[33moptional-dependencies[39;49;00m[33m'[39;49;00m][[33m'[39;49;00m[33mfoo[39;49;00m[33m'[39;49;00m].append([33m'[39;49;00m[33mbinary[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
        project.save_config(config)[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m project_path.as_cwd(env_vars={ConfigEnvVars.DATA: [96mstr[39;49;00m(data_path)}):[90m[39;49;00m
            result = hatch([90m[39;49;00m
                [33m'[39;49;00m[33mrun[39;49;00m[33m'[39;49;00m,[90m[39;49;00m
                [33m'[39;49;00m[33mpython[39;49;00m[33m'[39;49;00m,[90m[39;49;00m
                [33m'[39;49;00m[33m-c[39;49;00m[33m'[39;49;00m,[90m[39;49;00m
                [33m"[39;49;00m[33mimport binary,pathlib,sys;pathlib.Path([39;49;00m[33m'[39;49;00m[33mtest.txt[39;49;00m[33m'[39;49;00m[33m).write_text(str(binary.convert_units(1024)))[39;49;00m[33m"[39;49;00m,[90m[39;49;00m
            )[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
>       [94massert[39;49;00m result.output == helpers.dedent([90m[39;49;00m
    [90m        [39;49;00m[33m"""[39;49;00m
    [33m        Checking dependencies[39;49;00m
    [33m        Syncing dependencies[39;49;00m
    [33m        """[39;49;00m[90m[39;49;00m
        )[90m[39;49;00m
[1m[31mE       AssertionError: assert 'Checking dependencies\n' == 'Checking dep...ependencies\n'[0m
[1m[31mE         [0m
[1m[31mE           Checking dependencies[0m
[1m[31mE         - Syncing dependencies[0m

[1m[31m/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/cli/run/test_run.py[0m:425: AssertionError
[31m[1m________________________ test_dependency_hash_checking _________________________[0m
[gw8] darwin -- Python 3.12.7 /nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12

hatch = <tests.conftest.CliRunner object at 0x11cfdb170>
helpers = <module 'tests.helpers.helpers' from '/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/helpers/helpers.py'>
temp_dir = Path('/private/tmp/nix-build-hatch-1.13.0.drv-0/tmp60ya75m7')
config_file = <hatch.config.user.ConfigFile object at 0x11c47b9b0>
mocker = <pytest_mock.plugin.MockerFixture object at 0x11c47a990>

    [0m[37m@pytest[39;49;00m.mark.requires_internet[90m[39;49;00m
    [94mdef[39;49;00m [92mtest_dependency_hash_checking[39;49;00m(hatch, helpers, temp_dir, config_file, mocker):[90m[39;49;00m
        config_file.model.template.plugins[[33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m][[33m'[39;49;00m[33mtests[39;49;00m[33m'[39;49;00m] = [94mFalse[39;49;00m[90m[39;49;00m
        config_file.save()[90m[39;49;00m
    [90m[39;49;00m
        project_name = [33m'[39;49;00m[33mMy.App[39;49;00m[33m'[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m temp_dir.as_cwd():[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33mnew[39;49;00m[33m'[39;49;00m, project_name)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
    [90m[39;49;00m
        project_path = temp_dir / [33m'[39;49;00m[33mmy-app[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path = temp_dir / [33m'[39;49;00m[33mdata[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path.mkdir()[90m[39;49;00m
    [90m[39;49;00m
        project = Project(project_path)[90m[39;49;00m
        helpers.update_project_environment(project, [33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m, {[33m'[39;49;00m[33mskip-install[39;49;00m[33m'[39;49;00m: [94mTrue[39;49;00m, **project.config.envs[[33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m]})[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m project_path.as_cwd(env_vars={ConfigEnvVars.DATA: [96mstr[39;49;00m(data_path)}):[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33menv[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mcreate[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
        [94massert[39;49;00m result.output == helpers.dedent([90m[39;49;00m
    [90m        [39;49;00m[33m"""[39;49;00m
    [33m        Creating environment: default[39;49;00m
    [33m        Checking dependencies[39;49;00m
    [33m        """[39;49;00m[90m[39;49;00m
        )[90m[39;49;00m
    [90m[39;49;00m
        env_data_path = data_path / [33m'[39;49;00m[33menv[39;49;00m[33m'[39;49;00m / [33m'[39;49;00m[33mvirtual[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        [94massert[39;49;00m env_data_path.is_dir()[90m[39;49;00m
    [90m[39;49;00m
        project_data_path = env_data_path / project_path.name[90m[39;49;00m
        [94massert[39;49;00m project_data_path.is_dir()[90m[39;49;00m
    [90m[39;49;00m
        storage_dirs = [96mlist[39;49;00m(project_data_path.iterdir())[90m[39;49;00m
        [94massert[39;49;00m [96mlen[39;49;00m(storage_dirs) == [94m1[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        storage_path = storage_dirs[[94m0[39;49;00m][90m[39;49;00m
        [94massert[39;49;00m [96mlen[39;49;00m(storage_path.name) == [94m8[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        env_dirs = [96mlist[39;49;00m(storage_path.iterdir())[90m[39;49;00m
        [94massert[39;49;00m [96mlen[39;49;00m(env_dirs) == [94m1[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        env_path = env_dirs[[94m0[39;49;00m][90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m env_path.name == project_path.name[90m[39;49;00m
    [90m[39;49;00m
        project = Project(project_path)[90m[39;49;00m
        helpers.update_project_environment([90m[39;49;00m
            project, [33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m, {[33m'[39;49;00m[33mdependencies[39;49;00m[33m'[39;49;00m: [[33m'[39;49;00m[33mbinary[39;49;00m[33m'[39;49;00m], **project.config.envs[[33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m]}[90m[39;49;00m
        )[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m project_path.as_cwd(env_vars={ConfigEnvVars.DATA: [96mstr[39;49;00m(data_path)}):[90m[39;49;00m
            result = hatch([90m[39;49;00m
                [33m'[39;49;00m[33mrun[39;49;00m[33m'[39;49;00m,[90m[39;49;00m
                [33m'[39;49;00m[33mpython[39;49;00m[33m'[39;49;00m,[90m[39;49;00m
                [33m'[39;49;00m[33m-c[39;49;00m[33m'[39;49;00m,[90m[39;49;00m
                [33m"[39;49;00m[33mimport binary,pathlib,sys;pathlib.Path([39;49;00m[33m'[39;49;00m[33mtest.txt[39;49;00m[33m'[39;49;00m[33m).write_text(str(binary.convert_units(1024)))[39;49;00m[33m"[39;49;00m,[90m[39;49;00m
            )[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
>       [94massert[39;49;00m result.output == helpers.dedent([90m[39;49;00m
    [90m        [39;49;00m[33m"""[39;49;00m
    [33m        Checking dependencies[39;49;00m
    [33m        Syncing dependencies[39;49;00m
    [33m        """[39;49;00m[90m[39;49;00m
        )[90m[39;49;00m
[1m[31mE       AssertionError: assert 'Checking dependencies\n' == 'Checking dep...ependencies\n'[0m
[1m[31mE         [0m
[1m[31mE           Checking dependencies[0m
[1m[31mE         - Syncing dependencies[0m

[1m[31m/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/cli/run/test_run.py[0m:500: AssertionError
[31m[1m__________________________ test_sync_dependencies_uv ___________________________[0m
[gw12] darwin -- Python 3.12.7 /nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12

hatch = <tests.conftest.CliRunner object at 0x107c150d0>
helpers = <module 'tests.helpers.helpers' from '/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/helpers/helpers.py'>
temp_dir = Path('/private/tmp/nix-build-hatch-1.13.0.drv-0/tmpdtw5_rvv')
platform = <hatch.utils.platform.Platform object at 0x102696d80>
uv_on_path = '/nix/store/kcam2c7p2n7k6kazx8ipy9v93mmp2acd-uv-0.4.30/bin/uv'
extract_installed_requirements = <function extract_installed_requirements.<locals>.<lambda> at 0x10715d580>

    [0m[37m@pytest[39;49;00m.mark.requires_internet[90m[39;49;00m
    [94mdef[39;49;00m [92mtest_sync_dependencies_uv[39;49;00m(hatch, helpers, temp_dir, platform, uv_on_path, extract_installed_requirements):[90m[39;49;00m
        project_name = [33m'[39;49;00m[33mMy.App[39;49;00m[33m'[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m temp_dir.as_cwd():[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33mnew[39;49;00m[33m'[39;49;00m, project_name)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
    [90m[39;49;00m
        project_path = temp_dir / [33m'[39;49;00m[33mmy-app[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path = temp_dir / [33m'[39;49;00m[33mdata[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path.mkdir()[90m[39;49;00m
    [90m[39;49;00m
        project = Project(project_path)[90m[39;49;00m
        helpers.update_project_environment([90m[39;49;00m
            project,[90m[39;49;00m
            [33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m,[90m[39;49;00m
            {[90m[39;49;00m
                [33m'[39;49;00m[33mdependencies[39;49;00m[33m'[39;49;00m: [[33m'[39;49;00m[33mbinary[39;49;00m[33m'[39;49;00m],[90m[39;49;00m
                [33m'[39;49;00m[33mpost-install-commands[39;49;00m[33m'[39;49;00m: [[33m"[39;49;00m[33mpython -c [39;49;00m[33m\"[39;49;00m[33mwith open([39;49;00m[33m'[39;49;00m[33mtest.txt[39;49;00m[33m'[39;49;00m[33m, [39;49;00m[33m'[39;49;00m[33mw[39;49;00m[33m'[39;49;00m[33m) as f: f.write([39;49;00m[33m'[39;49;00m[33mcontent[39;49;00m[33m'[39;49;00m[33m)[39;49;00m[33m\"[39;49;00m[33m"[39;49;00m],[90m[39;49;00m
                **project.config.envs[[33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m],[90m[39;49;00m
            },[90m[39;49;00m
        )[90m[39;49;00m
        helpers.update_project_environment(project, [33m'[39;49;00m[33mtest[39;49;00m[33m'[39;49;00m, {})[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m project_path.as_cwd(env_vars={ConfigEnvVars.DATA: [96mstr[39;49;00m(data_path)}):[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33menv[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mcreate[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mtest[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
>       [94massert[39;49;00m result.output == helpers.dedent([90m[39;49;00m
    [90m        [39;49;00m[33m"""[39;49;00m
    [33m        Creating environment: test[39;49;00m
    [33m        Installing project in development mode[39;49;00m
    [33m        Running post-installation commands[39;49;00m
    [33m        Checking dependencies[39;49;00m
    [33m        Syncing dependencies[39;49;00m
    [33m        """[39;49;00m[90m[39;49;00m
        )[90m[39;49;00m
[1m[31mE       AssertionError: assert 'Creating env...ependencies\n' == 'Creating env...ependencies\n'[0m
[1m[31mE         [0m
[1m[31mE         Skipping 112 identical leading characters in diff, use -v to show[0m
[1m[31mE           pendencies[0m
[1m[31mE         - Syncing dependencies[0m

[1m[31m/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/cli/env/test_create.py[0m:1307: AssertionError
------------------------------ Captured log call -------------------------------
INFO     root:builtin.py:72 find interpreter for spec PythonSpec()
INFO     root:builtin.py:79 proposed PythonInfo(spec=CPython3.12.7.final.0-64, exe=/nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12, platform=darwin, version='3.12.7 (main, Oct  1 2024, 02:05:46) [Clang 16.0.6 ]', encoding_fs_io=utf-8-utf-8)
DEBUG    root:builtin.py:81 accepted PythonInfo(spec=CPython3.12.7.final.0-64, exe=/nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12, platform=darwin, version='3.12.7 (main, Oct  1 2024, 02:05:46) [Clang 16.0.6 ]', encoding_fs_io=utf-8-utf-8)
INFO     root:builtin.py:72 find interpreter for spec PythonSpec(major=3, minor=12)
INFO     root:builtin.py:79 proposed PythonInfo(spec=CPython3.12.7.final.0-64, exe=/nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12, platform=darwin, version='3.12.7 (main, Oct  1 2024, 02:05:46) [Clang 16.0.6 ]', encoding_fs_io=utf-8-utf-8)
DEBUG    root:builtin.py:81 accepted PythonInfo(spec=CPython3.12.7.final.0-64, exe=/nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12, platform=darwin, version='3.12.7 (main, Oct  1 2024, 02:05:46) [Clang 16.0.6 ]', encoding_fs_io=utf-8-utf-8)
[31m[1m__________________________ test_sync_dependencies_pip __________________________[0m
[gw12] darwin -- Python 3.12.7 /nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12

hatch = <tests.conftest.CliRunner object at 0x107c150d0>
helpers = <module 'tests.helpers.helpers' from '/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/helpers/helpers.py'>
temp_dir = Path('/private/tmp/nix-build-hatch-1.13.0.drv-0/tmpk68y84rj')
platform = <hatch.utils.platform.Platform object at 0x102696d80>
extract_installed_requirements = <function extract_installed_requirements.<locals>.<lambda> at 0x10715d580>

    [0m[37m@pytest[39;49;00m.mark.requires_internet[90m[39;49;00m
    [94mdef[39;49;00m [92mtest_sync_dependencies_pip[39;49;00m(hatch, helpers, temp_dir, platform, extract_installed_requirements):[90m[39;49;00m
        project_name = [33m'[39;49;00m[33mMy.App[39;49;00m[33m'[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m temp_dir.as_cwd():[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33mnew[39;49;00m[33m'[39;49;00m, project_name)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
    [90m[39;49;00m
        project_path = temp_dir / [33m'[39;49;00m[33mmy-app[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path = temp_dir / [33m'[39;49;00m[33mdata[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path.mkdir()[90m[39;49;00m
    [90m[39;49;00m
        project = Project(project_path)[90m[39;49;00m
        helpers.update_project_environment([90m[39;49;00m
            project,[90m[39;49;00m
            [33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m,[90m[39;49;00m
            {[90m[39;49;00m
                [33m'[39;49;00m[33mdependencies[39;49;00m[33m'[39;49;00m: [[33m'[39;49;00m[33mbinary[39;49;00m[33m'[39;49;00m],[90m[39;49;00m
                [33m'[39;49;00m[33mpost-install-commands[39;49;00m[33m'[39;49;00m: [[33m"[39;49;00m[33mpython -c [39;49;00m[33m\"[39;49;00m[33mwith open([39;49;00m[33m'[39;49;00m[33mtest.txt[39;49;00m[33m'[39;49;00m[33m, [39;49;00m[33m'[39;49;00m[33mw[39;49;00m[33m'[39;49;00m[33m) as f: f.write([39;49;00m[33m'[39;49;00m[33mcontent[39;49;00m[33m'[39;49;00m[33m)[39;49;00m[33m\"[39;49;00m[33m"[39;49;00m],[90m[39;49;00m
                **project.config.envs[[33m'[39;49;00m[33mdefault[39;49;00m[33m'[39;49;00m],[90m[39;49;00m
            },[90m[39;49;00m
        )[90m[39;49;00m
        helpers.update_project_environment(project, [33m'[39;49;00m[33mtest[39;49;00m[33m'[39;49;00m, {})[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m project_path.as_cwd(), EnvVars([90m[39;49;00m
            {ConfigEnvVars.DATA: [96mstr[39;49;00m(data_path)}, exclude=[get_env_var(plugin_name=[33m'[39;49;00m[33mvirtual[39;49;00m[33m'[39;49;00m, option=[33m'[39;49;00m[33muv_path[39;49;00m[33m'[39;49;00m)][90m[39;49;00m
        ):[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33menv[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mcreate[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mtest[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
>       [94massert[39;49;00m result.output == helpers.dedent([90m[39;49;00m
    [90m        [39;49;00m[33m"""[39;49;00m
    [33m        Creating environment: test[39;49;00m
    [33m        Installing project in development mode[39;49;00m
    [33m        Running post-installation commands[39;49;00m
    [33m        Checking dependencies[39;49;00m
    [33m        Syncing dependencies[39;49;00m
    [33m        """[39;49;00m[90m[39;49;00m
        )[90m[39;49;00m
[1m[31mE       AssertionError: assert 'Creating env...ependencies\n' == 'Creating env...ependencies\n'[0m
[1m[31mE         [0m
[1m[31mE         Skipping 112 identical leading characters in diff, use -v to show[0m
[1m[31mE           pendencies[0m
[1m[31mE         - Syncing dependencies[0m

[1m[31m/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/cli/env/test_create.py[0m:1379: AssertionError
------------------------------ Captured log call -------------------------------
INFO     root:builtin.py:72 find interpreter for spec PythonSpec()
INFO     root:builtin.py:79 proposed PythonInfo(spec=CPython3.12.7.final.0-64, exe=/nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12, platform=darwin, version='3.12.7 (main, Oct  1 2024, 02:05:46) [Clang 16.0.6 ]', encoding_fs_io=utf-8-utf-8)
DEBUG    root:builtin.py:81 accepted PythonInfo(spec=CPython3.12.7.final.0-64, exe=/nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12, platform=darwin, version='3.12.7 (main, Oct  1 2024, 02:05:46) [Clang 16.0.6 ]', encoding_fs_io=utf-8-utf-8)
INFO     root:builtin.py:72 find interpreter for spec PythonSpec(major=3, minor=12)
INFO     root:builtin.py:79 proposed PythonInfo(spec=CPython3.12.7.final.0-64, exe=/nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12, platform=darwin, version='3.12.7 (main, Oct  1 2024, 02:05:46) [Clang 16.0.6 ]', encoding_fs_io=utf-8-utf-8)
DEBUG    root:builtin.py:81 accepted PythonInfo(spec=CPython3.12.7.final.0-64, exe=/nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12, platform=darwin, version='3.12.7 (main, Oct  1 2024, 02:05:46) [Clang 16.0.6 ]', encoding_fs_io=utf-8-utf-8)
[31m[1m______________________ TestScriptRunner.test_dependencies ______________________[0m
[gw1] darwin -- Python 3.12.7 /nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12

self = <tests.cli.run.test_run.TestScriptRunner object at 0x105ac1850>
hatch = <tests.conftest.CliRunner object at 0x109f20500>
helpers = <module 'tests.helpers.helpers' from '/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/helpers/helpers.py'>
temp_dir = Path('/private/tmp/nix-build-hatch-1.13.0.drv-0/tmp8sptjaiq')

    [0m[37m@pytest[39;49;00m.mark.requires_internet[90m[39;49;00m
    [94mdef[39;49;00m [92mtest_dependencies[39;49;00m([96mself[39;49;00m, hatch, helpers, temp_dir):[90m[39;49;00m
        data_path = temp_dir / [33m'[39;49;00m[33mdata[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path.mkdir()[90m[39;49;00m
        script = (temp_dir / [33m'[39;49;00m[33mscript.py[39;49;00m[33m'[39;49;00m).resolve()[90m[39;49;00m
        script.write_text([90m[39;49;00m
            helpers.dedent([90m[39;49;00m
    [90m            [39;49;00m[33m"""[39;49;00m
    [33m            # /// script[39;49;00m
    [33m            # dependencies = [[39;49;00m
    [33m            #   "binary",[39;49;00m
    [33m            # ][39;49;00m
    [33m            # ///[39;49;00m
    [33m            import pathlib[39;49;00m
    [33m            import sys[39;49;00m
    [33m[39;49;00m
    [33m            import binary[39;49;00m
    [33m[39;49;00m
    [33m            pathlib.Path('test.txt').write_text([39;49;00m
    [33m                f'{sys.executable}\\n{str(binary.convert_units(1024))}'[39;49;00m
    [33m            )[39;49;00m
    [33m            """[39;49;00m[90m[39;49;00m
            )[90m[39;49;00m
        )[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m temp_dir.as_cwd(env_vars={ConfigEnvVars.DATA: [96mstr[39;49;00m(data_path)}):[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33mrun[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mscript.py[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
>       [94massert[39;49;00m result.output == helpers.dedent([90m[39;49;00m
            [33mf[39;49;00m[33m"""[39;49;00m[33m[39;49;00m
    [33m        Creating environment: [39;49;00m[33m{[39;49;00mscript.id[33m}[39;49;00m[33m[39;49;00m
    [33m        Checking dependencies[39;49;00m[33m[39;49;00m
    [33m        Syncing dependencies[39;49;00m[33m[39;49;00m
    [33m        [39;49;00m[33m"""[39;49;00m[90m[39;49;00m
        )[90m[39;49;00m
[1m[31mE       AssertionError: assert 'Creating env...ependencies\n' == 'Creating env...ependencies\n'[0m
[1m[31mE         [0m
[1m[31mE         Skipping 42 identical leading characters in diff, use -v to show[0m
[1m[31mE           pendencies[0m
[1m[31mE         - Syncing dependencies[0m

[1m[31m/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/cli/run/test_run.py[0m:2411: AssertionError
[31m[1m_____________ TestScriptRunner.test_dependencies_from_tool_config ______________[0m
[gw1] darwin -- Python 3.12.7 /nix/store/7c494qcmh62av43zsxr3wvzh8hcpy1vl-python3-3.12.7/bin/python3.12

self = <tests.cli.run.test_run.TestScriptRunner object at 0x105ac1c40>
hatch = <tests.conftest.CliRunner object at 0x109f20500>
helpers = <module 'tests.helpers.helpers' from '/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/helpers/helpers.py'>
temp_dir = Path('/private/tmp/nix-build-hatch-1.13.0.drv-0/tmp2zqvugeq')

    [0m[37m@pytest[39;49;00m.mark.requires_internet[90m[39;49;00m
    [94mdef[39;49;00m [92mtest_dependencies_from_tool_config[39;49;00m([96mself[39;49;00m, hatch, helpers, temp_dir):[90m[39;49;00m
        data_path = temp_dir / [33m'[39;49;00m[33mdata[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        data_path.mkdir()[90m[39;49;00m
        script = (temp_dir / [33m'[39;49;00m[33mscript.py[39;49;00m[33m'[39;49;00m).resolve()[90m[39;49;00m
        script.write_text([90m[39;49;00m
            helpers.dedent([90m[39;49;00m
    [90m            [39;49;00m[33m"""[39;49;00m
    [33m            # /// script[39;49;00m
    [33m            # dependencies = [][39;49;00m
    [33m            #[39;49;00m
    [33m            # [tool.hatch][39;49;00m
    [33m            # dependencies = [[39;49;00m
    [33m            #   "binary",[39;49;00m
    [33m            # ][39;49;00m
    [33m            # ///[39;49;00m
    [33m            import pathlib[39;49;00m
    [33m            import sys[39;49;00m
    [33m[39;49;00m
    [33m            import binary[39;49;00m
    [33m[39;49;00m
    [33m            pathlib.Path('test.txt').write_text([39;49;00m
    [33m                f'{sys.executable}\\n{str(binary.convert_units(1024))}'[39;49;00m
    [33m            )[39;49;00m
    [33m            """[39;49;00m[90m[39;49;00m
            )[90m[39;49;00m
        )[90m[39;49;00m
    [90m[39;49;00m
        [94mwith[39;49;00m temp_dir.as_cwd(env_vars={ConfigEnvVars.DATA: [96mstr[39;49;00m(data_path)}):[90m[39;49;00m
            result = hatch([33m'[39;49;00m[33mrun[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33mscript.py[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m result.exit_code == [94m0[39;49;00m, result.output[90m[39;49;00m
>       [94massert[39;49;00m result.output == helpers.dedent([90m[39;49;00m
            [33mf[39;49;00m[33m"""[39;49;00m[33m[39;49;00m
    [33m        Creating environment: [39;49;00m[33m{[39;49;00mscript.id[33m}[39;49;00m[33m[39;49;00m
    [33m        Checking dependencies[39;49;00m[33m[39;49;00m
    [33m        Syncing dependencies[39;49;00m[33m[39;49;00m
    [33m        [39;49;00m[33m"""[39;49;00m[90m[39;49;00m
        )[90m[39;49;00m
[1m[31mE       AssertionError: assert 'Creating env...ependencies\n' == 'Creating env...ependencies\n'[0m
[1m[31mE         [0m
[1m[31mE         Skipping 42 identical leading characters in diff, use -v to show[0m
[1m[31mE           pendencies[0m
[1m[31mE         - Syncing dependencies[0m

[1m[31m/private/tmp/nix-build-hatch-1.13.0.drv-0/source/tests/cli/run/test_run.py[0m:2467: AssertionError
[36m[1m=========================== short test summary info ============================[0m
[31mFAILED[0m tests/cli/project/test_metadata.py::[1mTestBuildDependenciesMissing::test_no_compatibility_check_if_exists[0m - AssertionError: assert 'MIT\n' == 'Setting up b...encies\nMIT\n'
[31mFAILED[0m tests/cli/run/test_run.py::[1mtest_sync_dependencies[0m - AssertionError: assert 'Checking dependencies\n' == 'Checking dep...ependen...
[31mFAILED[0m tests/cli/version/test_version.py::[1mtest_no_compatibility_check_if_exists[0m - AssertionError: assert '0.0.1\n' == 'Setting up b...ependencies\n'
[31mFAILED[0m tests/cli/run/test_run.py::[1mtest_sync_project_dependencies[0m - AssertionError: assert 'Checking dependencies\n' == 'Checking dep...ependen...
[31mFAILED[0m tests/cli/run/test_run.py::[1mtest_sync_project_features[0m - AssertionError: assert 'Checking dependencies\n' == 'Checking dep...ependen...
[31mFAILED[0m tests/cli/run/test_run.py::[1mtest_dependency_hash_checking[0m - AssertionError: assert 'Checking dependencies\n' == 'Checking dep...ependen...
[31mFAILED[0m tests/cli/env/test_create.py::[1mtest_sync_dependencies_uv[0m - AssertionError: assert 'Creating env...ependencies\n' == 'Creating env...ep...
[31mFAILED[0m tests/cli/env/test_create.py::[1mtest_sync_dependencies_pip[0m - AssertionError: assert 'Creating env...ependencies\n' == 'Creating env...ep...
[31mFAILED[0m tests/cli/run/test_run.py::[1mTestScriptRunner::test_dependencies[0m - AssertionError: assert 'Creating env...ependencies\n' == 'Creating env...ep...
[31mFAILED[0m tests/cli/run/test_run.py::[1mTestScriptRunner::test_dependencies_from_tool_config[0m - AssertionError: assert 'Creating env...ependencies\n' == 'Creating env...ep...
[31m================= [31m[1m10 failed[0m, [32m1490 passed[0m, [33m53 skipped[0m[31m in 28.57s[0m[31m =================[0m
```

</details>

## Metadata

 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.1.0, macOS 15.1.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.24.10`
 - nixpkgs: `/nix/store/22r7q7s9552gn1vpjigkbhfgcvhsrz68-source`

## Notify maintainers

@onny

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

> > I am able to recreate this on nixos-unstable on nixos
> 
> I'm also getting a build (pytestCheckPhase) error (on nixos x86_64-linux), since `github:nixos/nixpkgs/634fd46801442d760e09493a794c4f15db2d0cbb`. End of build log says a bunch of tests, separate from the above, failed:
> 
> ```
> =========================== short test summary info ============================
> FAILED tests/backend/builders/test_sdist.py::TestBuildStandard::test_include_project_file - AssertionError: my_app-0.0.1/PKG-INFO
> FAILED tests/backend/builders/test_sdist.py::TestBuildStandard::test_project_file_always_included - AssertionError: my_app-0.0.1/PKG-INFO
> FAILED tests/backend/builders/test_sdist.py::TestBuildStandard::test_config_file_always_included - AssertionError: my_app-0.0.1/PKG-INFO
> FAILED tests/backend/builders/test_sdist.py::TestBuildStandard::test_include_readme - AssertionError: my_app-0.0.1/PKG-INFO
> FAILED tests/backend/metadata/test_spec.py::TestCoreMetadataV23::test_all[construct_metadata_file_2_3] - AssertionError: assert 'Metadata-Ver...est content\n' == 'Metadata-Ver...es...
> FAILED tests/backend/builders/test_sdist.py::TestBuildStandard::test_readme_always_included - AssertionError: my_app-0.0.1/PKG-INFO
> FAILED tests/backend/builders/test_sdist.py::TestBuildStandard::test_include_license_files - AssertionError: my_app-0.0.1/PKG-INFO
> FAILED tests/backend/builders/test_sdist.py::TestBuildStandard::test_license_files_always_included - AssertionError: my_app-0.0.1/PKG-INFO
> FAILED tests/backend/builders/test_sdist.py::TestBuildStandard::test_no_strict_naming - AssertionError: My.App-0.0.1/PKG-INFO
> FAILED tests/backend/metadata/test_spec.py::TestProjectMetadataFromCoreMetadata::test_license_files - AssertionError: assert {'license-fil...ion': '0.1.0'} == {'license-fil...io...
> FAILED tests/backend/builders/test_wheel.py::TestBuildStandard::test_explicit_selection_with_src_layout - AssertionError: my_app-0.0.1.dist-info/METADATA
> FAILED tests/backend/builders/test_wheel.py::TestBuildStandard::test_single_module - AssertionError: my_app-0.0.1.dist-info/METADATA
> FAILED tests/backend/metadata/test_spec.py::TestCoreMetadataV23::test_license_expression[construct_metadata_file_2_3] - AssertionError: assert 'Metadata-Ver...sion: 0.1.0\n' == 'Metadata-Ver... A...
> FAILED tests/backend/metadata/test_spec.py::TestCoreMetadataV23::test_license_files[construct_metadata_file_2_3] - AssertionError: assert 'Metadata-Ver...sion: 0.1.0\n' == 'Metadata-Ver...SE...
> FAILED tests/backend/builders/test_wheel.py::TestBuildStandard::test_no_strict_naming - AssertionError: My.App-0.0.1.dist-info/METADATA
> ERROR tests/backend/licenses/test_parse.py
> ERROR tests/backend/licenses/test_supported.py
> ============ 15 failed, 1474 passed, 116 skipped, 2 errors in 6.81s ============
> ```

which hatch version is this? there is a new version on master since some hours ago

> > > I am able to recreate this on nixos-unstable on nixos
> > 
> > 
> > I'm also getting a build (pytestCheckPhase) error (on nixos x86_64-linux), since `github:nixos/nixpkgs/634fd46801442d760e09493a794c4f15db2d0cbb`. End of build log says a bunch of tests, separate from the above, failed:
> > ```
> > =========================== short test summary info ============================
> > FAILED tests/backend/builders/test_sdist.py::TestBuildStandard::test_include_project_file - AssertionError: my_app-0.0.1/PKG-INFO
> > FAILED tests/backend/builders/test_sdist.py::TestBuildStandard::test_project_file_always_included - AssertionError: my_app-0.0.1/PKG-INFO
> > FAILED tests/backend/builders/test_sdist.py::TestBuildStandard::test_config_file_always_included - AssertionError: my_app-0.0.1/PKG-INFO
> > FAILED tests/backend/builders/test_sdist.py::TestBuildStandard::test_include_readme - AssertionError: my_app-0.0.1/PKG-INFO
> > FAILED tests/backend/metadata/test_spec.py::TestCoreMetadataV23::test_all[construct_metadata_file_2_3] - AssertionError: assert 'Metadata-Ver...est content\n' == 'Metadata-Ver...es...
> > FAILED tests/backend/builders/test_sdist.py::TestBuildStandard::test_readme_always_included - AssertionError: my_app-0.0.1/PKG-INFO
> > FAILED tests/backend/builders/test_sdist.py::TestBuildStandard::test_include_license_files - AssertionError: my_app-0.0.1/PKG-INFO
> > FAILED tests/backend/builders/test_sdist.py::TestBuildStandard::test_license_files_always_included - AssertionError: my_app-0.0.1/PKG-INFO
> > FAILED tests/backend/builders/test_sdist.py::TestBuildStandard::test_no_strict_naming - AssertionError: My.App-0.0.1/PKG-INFO
> > FAILED tests/backend/metadata/test_spec.py::TestProjectMetadataFromCoreMetadata::test_license_files - AssertionError: assert {'license-fil...ion': '0.1.0'} == {'license-fil...io...
> > FAILED tests/backend/builders/test_wheel.py::TestBuildStandard::test_explicit_selection_with_src_layout - AssertionError: my_app-0.0.1.dist-info/METADATA
> > FAILED tests/backend/builders/test_wheel.py::TestBuildStandard::test_single_module - AssertionError: my_app-0.0.1.dist-info/METADATA
> > FAILED tests/backend/metadata/test_spec.py::TestCoreMetadataV23::test_license_expression[construct_metadata_file_2_3] - AssertionError: assert 'Metadata-Ver...sion: 0.1.0\n' == 'Metadata-Ver... A...
> > FAILED tests/backend/metadata/test_spec.py::TestCoreMetadataV23::test_license_files[construct_metadata_file_2_3] - AssertionError: assert 'Metadata-Ver...sion: 0.1.0\n' == 'Metadata-Ver...SE...
> > FAILED tests/backend/builders/test_wheel.py::TestBuildStandard::test_no_strict_naming - AssertionError: My.App-0.0.1.dist-info/METADATA
> > ERROR tests/backend/licenses/test_parse.py
> > ERROR tests/backend/licenses/test_supported.py
> > ============ 15 failed, 1474 passed, 116 skipped, 2 errors in 6.81s ============
> > ```
> 
> which hatch version is this? there is a new version on master since some hours ago

Attempting to build from latest staging, where the pr was merged, or the commit it was merged in, errors on testing `hyq1gv9y1ryp8mp00wmh1ms343ajpr7h-python3.12-fs-2.4.16` (same derivation failing).
```
=========================== short test summary info ============================
FAILED tests/test_encoding.py::TestEncoding::test_listdir - OSError: [Errno 84] Invalid or incomplete multibyte or wide character: '/bu...
FAILED tests/test_encoding.py::TestEncoding::test_open - OSError: [Errno 84] Invalid or incomplete multibyte or wide character: '/bu...
FAILED tests/test_encoding.py::TestEncoding::test_scandir - OSError: [Errno 84] Invalid or incomplete multibyte or wide character: '/bu...
=== 3 failed, 2041 passed, 24 skipped, 272 deselected, 18 warnings in 10.53s ===
```

So, I am unsure if hatch is failing now.

`hatch` is still failing to build from staging on my computer (with zfs) as of `10ba47da060314479cb7c4cbb766ad5bb02fbbf3`, with the exact same test failures mentioned above for `python3.12-fs-2.4.16` (maintained by @lovek323 ).

<details>

```
=================================== FAILURES ===================================
__________________________ TestEncoding.test_listdir ___________________________

self = <tests.test_encoding.TestEncoding testMethod=test_listdir>

    def setUp(self):
        dir_path = self.dir_path = tempfile.mkdtemp()
        if six.PY2:
            with open(os.path.join(dir_path, self.TEST_FILENAME), "wb") as f:
                f.write(b"baz")
        else:
>           with open(
                os.path.join(dir_path, self.TEST_FILENAME_UNICODE), "wb"
            ) as f:
E           OSError: [Errno 84] Invalid or incomplete multibyte or wide character: '/build/tmpv6ec1jxe/foo\udcb1bar'

tests/test_encoding.py:28: OSError
____________________________ TestEncoding.test_open ____________________________

self = <tests.test_encoding.TestEncoding testMethod=test_open>

    def setUp(self):
        dir_path = self.dir_path = tempfile.mkdtemp()
        if six.PY2:
            with open(os.path.join(dir_path, self.TEST_FILENAME), "wb") as f:
                f.write(b"baz")
        else:
>           with open(
                os.path.join(dir_path, self.TEST_FILENAME_UNICODE), "wb"
            ) as f:
E           OSError: [Errno 84] Invalid or incomplete multibyte or wide character: '/build/tmpr9_7s7it/foo\udcb1bar'

tests/test_encoding.py:28: OSError
__________________________ TestEncoding.test_scandir ___________________________

self = <tests.test_encoding.TestEncoding testMethod=test_scandir>

    def setUp(self):
        dir_path = self.dir_path = tempfile.mkdtemp()
        if six.PY2:
            with open(os.path.join(dir_path, self.TEST_FILENAME), "wb") as f:
                f.write(b"baz")
        else:
>           with open(
                os.path.join(dir_path, self.TEST_FILENAME_UNICODE), "wb"
            ) as f:
E           OSError: [Errno 84] Invalid or incomplete multibyte or wide character: '/build/tmpmbmrz8vq/foo\udcb1bar'

tests/test_encoding.py:28: OSError
=============================== warnings summary ===============================
fs/__init__.py:4
  /build/fs-2.4.16/fs/__init__.py:4: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    __import__("pkg_resources").declare_namespace(__name__)  # type: ignore

fs/__init__.py:4
tests/test_imports.py::TestImports::test_import_path
  /build/fs-2.4.16/fs/__init__.py:4: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('fs')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    __import__("pkg_resources").declare_namespace(__name__)  # type: ignore

fs/opener/__init__.py:6
  /build/fs-2.4.16/fs/opener/__init__.py:6: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('fs.opener')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    __import__("pkg_resources").declare_namespace(__name__)  # type: ignore

../../nix/store/bm6i6h0s60k09b9gz15jj0lfv9q9q8yl-python3.12-setuptools-75.3.0/lib/python3.12/site-packages/pkg_resources/__init__.py:2563
  /nix/store/bm6i6h0s60k09b9gz15jj0lfv9q9q8yl-python3.12-setuptools-75.3.0/lib/python3.12/site-packages/pkg_resources/__init__.py:2563: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('fs')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(parent)

fs/base.py:75
fs/base.py:75
fs/base.py:75
fs/base.py:75
fs/base.py:75
  /build/fs-2.4.16/fs/base.py:75: DeprecationWarning: method 'settext' has been deprecated, please rename to 'writetext'
    warnings.warn(

../../nix/store/jg4wiya54y89yhsdrp1wlngln07svmgv-python3.12-pyftpdlib-1.5.10/lib/python3.12/site-packages/pyftpdlib/authorizers.py:408
  /nix/store/jg4wiya54y89yhsdrp1wlngln07svmgv-python3.12-pyftpdlib-1.5.10/lib/python3.12/site-packages/pyftpdlib/authorizers.py:408: DeprecationWarning: 'crypt' is deprecated and slated for removal in Python 3.13
    import crypt

../../nix/store/jg4wiya54y89yhsdrp1wlngln07svmgv-python3.12-pyftpdlib-1.5.10/lib/python3.12/site-packages/pyftpdlib/authorizers.py:410
  /nix/store/jg4wiya54y89yhsdrp1wlngln07svmgv-python3.12-pyftpdlib-1.5.10/lib/python3.12/site-packages/pyftpdlib/authorizers.py:410: DeprecationWarning: 'spwd' is deprecated and slated for removal in Python 3.13
    import spwd

tests/test_ftpfs.py::TestAnonFTPFS::test_appendbytes
  /nix/store/jg4wiya54y89yhsdrp1wlngln07svmgv-python3.12-pyftpdlib-1.5.10/lib/python3.12/site-packages/pyftpdlib/authorizers.py:118: RuntimeWarning: write permissions assigned to anonymous user.
    self._check_permissions(username, perm)

tests/test_ftpfs.py::TestAnonFTPFS::test_setinfo
tests/test_ftpfs.py::TestAnonFTPFS::test_settimes
tests/test_ftpfs.py::TestAnonFTPFS::test_touch
  /build/fs-2.4.16/fs/ftpfs.py:868: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    + datetime.datetime.utcfromtimestamp(mtime).strftime("%Y%m%d%H%M%S")

tests/test_wrap.py::TestWrapReadOnlySyspath::test_copy_file_if_newer
  /build/fs-2.4.16/fs/copy.py:163: DeprecationWarning: copy_file_if_newer is deprecated. Use copy_file_if instead.
    warnings.warn(

tests/test_wrap.py::TestWrapReadOnlySyspath::test_copy_fs_if_newer
  /build/fs-2.4.16/fs/copy.py:70: DeprecationWarning: copy_fs_if_newer is deprecated. Use copy_fs_if instead.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_encoding.py::TestEncoding::test_listdir - OSError: [Errno 84] Invalid or incomplete multibyte or wide character: '/bu...
FAILED tests/test_encoding.py::TestEncoding::test_open - OSError: [Errno 84] Invalid or incomplete multibyte or wide character: '/bu...
FAILED tests/test_encoding.py::TestEncoding::test_scandir - OSError: [Errno 84] Invalid or incomplete multibyte or wide character: '/bu...
=== 3 failed, 2041 passed, 24 skipped, 272 deselected, 18 warnings in 19.35s ===
```

</details>

--- a/pkgs/by-name/ha/hatch/package.nix
+++ b/pkgs/by-name/ha/hatch/package.nix
@@ -74,6 +74,19 @@ python3Packages.buildPythonApplication rec {
     export HOME=$(mktemp -d);
   '';
 
+  pytestFlagsArray = [
+    # AssertionError on the version metadata
+    # https://github.com/pypa/hatch/issues/1877
+    "--deselect=tests/backend/metadata/test_spec.py::TestCoreMetadataV21::test_all"
+    "--deselect=tests/backend/metadata/test_spec.py::TestCoreMetadataV21::test_license_expression"
+    "--deselect=tests/backend/metadata/test_spec.py::TestCoreMetadataV22::test_all"
+    "--deselect=tests/backend/metadata/test_spec.py::TestCoreMetadataV22::test_license_expression"
+    "--deselect=tests/backend/metadata/test_spec.py::TestCoreMetadataV23::test_all"
+    "--deselect=tests/backend/metadata/test_spec.py::TestCoreMetadataV23::test_license_expression"
+    "--deselect=tests/backend/metadata/test_spec.py::TestCoreMetadataV23::test_license_files"
+    "--deselect=tests/backend/metadata/test_spec.py::TestProjectMetadataFromCoreMetadata::test_license_files"
+  ];
+
   disabledTests =
     [
       # AssertionError: assert (1980, 1, 2, 0, 0, 0) == (2020, 2, 2, 0, 0, 0)
@@ -117,14 +130,21 @@ python3Packages.buildPythonApplication rec {
     ]
     ++ lib.optionals stdenv.hostPlatform.isAarch64 [ "test_resolve" ];
 
-  disabledTestPaths = lib.optionals stdenv.hostPlatform.isDarwin [
-    # AssertionError: assert [call('test h...2p32/bin/sh')] == [call('test h..., shell=True)]
-    # At index 0 diff:
-    #    call('test hatch-test.py3.10', shell=True, executable='/nix/store/b34ianga4diikh0kymkpqwmvba0mmzf7-bash-5.2p32/bin/sh')
-    # != call('test hatch-test.py3.10', shell=True)
-    "tests/cli/fmt/test_fmt.py"
-    "tests/cli/test/test_test.py"
-  ];
+  disabledTestPaths =
+    [
+      # ModuleNotFoundError: No module named 'hatchling.licenses.parse'
+      # https://github.com/pypa/hatch/issues/1850
+      "tests/backend/licenses/test_parse.py"
+      "tests/backend/licenses/test_supported.py"
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      # AssertionError: assert [call('test h...2p32/bin/sh')] == [call('test h..., shell=True)]
+      # At index 0 diff:
+      #    call('test hatch-test.py3.10', shell=True, executable='/nix/store/b34ianga4diikh0kymkpqwmvba0mmzf7-bash-5.2p32/bin/sh')
+      # != call('test hatch-test.py3.10', shell=True)
+      "tests/cli/fmt/test_fmt.py"
+      "tests/cli/test/test_test.py"
+    ];
 
   passthru = {
     updateScript = nix-update-script {

